packages/web/src/application/test/zoom/zoom-provider.test.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { ReactNode } from 'react';
+import { MemoryRouter } from 'react-router-dom';
+import { render, screen, waitFor } from '@testing-library/react';
+import { User } from '@chess-tent/models';
+import application from '@application';
+import { users, zoom } from '../fixtures';
+
+const { components } = application;
+
+interface RenderOptions {
+  children: ReactNode;
+  user: User;
+  meetingNumber?: string;
+  queryCode?: boolean;
+}
+
+const renderInsideProvider = ({
+  children,
+  user,
+  meetingNumber,
+  queryCode,
+}: RenderOptions) => {
+  const { ZoomProvider } = components;
+
+  render(
+    <MemoryRouter
+      initialEntries={[
+        { pathname: '/', search: queryCode ? '?code=testcode' : undefined },
+      ]}
+    >
+      <ZoomProvider
+        redirectUri={zoom.redirectUri}
+        user={user}
+        meetingNumber={meetingNumber}
+      >
+        {children}
+      </ZoomProvider>
+    </MemoryRouter>,
+  );
+};
+
+describe('Zoom Provider', () => {
+  const coach: User = users.find(user => user.coach)!;
+  const student: User = users.find(user => !user.coach)!;
+
+  test('Student should have only password input', async () => {
+    const { ZoomGuestControl } = components;
+
+    renderInsideProvider({
+      children: <ZoomGuestControl />,
+      user: student,
+      meetingNumber: zoom.meetingNumber,
+    });
+
+    const passwordInput = screen.queryByPlaceholderText(
+      'Meeting password (if any)',
+    );
+    const meetingNumberInput = screen.queryByPlaceholderText('Meeting number');
+
+    await waitFor(() => expect(passwordInput).toBeTruthy());
```

Comment:
You shouldn't need `waitFor` in this case? Do you have issues without it?

Avoid as much as possible use of `waitFor`.

Reply:
Yes, many issues, tried many alternatives but in the end this prevailed as the solution.
This was the first issue: https://stackoverflow.com/questions/55181009/jest-react-testing-library-warning-update-was-not-wrapped-in-act

But wrapping in act didn't help, it caused other issues, and found that it is also not recommended.

Reply:
Will take a look today

Reply:
This is happening because of the way the `useApi` hook is working. We have to go with waitFor, but there is a different way.

https://testing-library.com/docs/dom-testing-library/api-async/

`expect(await screen.findByText())...`

Use this approach for ZoomProvider, don't use it as a default way.